cki_lib/libcki.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+#
+# Common helper functions and global variables to be used by all CKI tests
+#
+# NOTE: To make coding style consistent, conventions in the following should be
+#       followed:
+#       1) All helper functions start with 'cki_';
+#       2) All global variables start with 'CKI_'.
+#
+
+source /usr/bin/rhts_environment.sh
+source /usr/share/beakerlib/beakerlib.sh
+
+CKI_RC_POS=0
+CKI_RC_NEG="1-255"
+CKI_RC_ANY="0-255" # To assist rlRun() to support any return code
+
+# Result code definitions
+CKI_PASS=0        # should go to rlPass()
+CKI_FAIL=1        # should go to rlFail()
+CKI_UNSUPPORTED=2 # should go to cki_skip_task()
+CKI_UNINITIATED=3 # should go to cki_abort_task()
+
+# Status code definitions
+CKI_STATUS_COMPLETED=0 # task is completed
+CKI_STATUS_ABORTED=1   # task is aborted
+
+#
+# When a serious problem occurs and we cannot proceed any further, we abort
+# this recipe with an error message.
+#
+# Arguments:
+#   $1 - the message to print in the log
+#   $2 - 'WARN' or 'FAIL'
+#
+function cki_abort_recipe()
+{
+    typeset failure_message=$1
+    typeset failure_type=${2:-"FAIL"}
+
+    echo "❌ ${failure_message}"
+    if [[ $failure_type == 'WARN' ]]; then
+        report_result ${TEST} WARN 99
+    else
+        report_result ${TEST} FAIL 1
+    fi
+    rhts-abort -t recipe
+    exit $CKI_STATUS_ABORTED
+}
+
+function cki_abort_task()
+{
+    rlLog "Aborting current task because $*"
+    rhts-report-result "$TEST" WARN "$OUTPUTFILE"
+    exit $CKI_STATUS_ABORTED
+}
+
+function cki_skip_task()
+{
+    rlLog "Skipping current task because $*"
+    rhts-report-result "$TEST" SKIP "$OUTPUTFILE"
+    exit $CKI_STATUS_COMPLETED
+}
+
+function cki_report_result
+{
+    typeset rc=${1?"*** result code"}
+    shift
+    typeset argv="$*"
+    case $rc in
+    $CKI_PASS) rlPass $argv ;;
+    $CKI_FAIL) rlFail "$g_reason_fail #$argv#" ;;
+    $CKI_UNSUPPORTED) cki_skip_task  "$g_reason_unsupported #$argv#" ;;
+    $CKI_UNINITIATED) cki_abort_task "$g_reason_uninitiated #$argv#" ;;
+    *) ;;
+    esac
+}
+
+#
+# Set reason for according to result code, once function cki_report_result() is
+# invoked, the related reason will be used when calling rlLog()
+#
+function cki_set_reason()
+{
+    typeset rc=${1?"*** result code"}
+    shift
+    case $rc in
+    $CKI_FAIL) g_reason_fail="$@" ;;
+    $CKI_UNSUPPORTED) g_reason_unsupported="$@" ;;
+    $CKI_UNINITIATED) g_reason_uninitiated="$@" ;;
+    *) ;;
```

Comment:
Maybe not ignore unexpected values in here, either?

Reply:
Added `*) g_reason_other="$rc is an invalid result code" ;;`.